content/blog/2022/100-percent-test-coverage.md

```diff
@@ -0,0 +1,136 @@
+---
+title: What's the impact of aiming for 100% test coverage?
+author: olu
+date: 2022-05-24
+tags:
+  - Article
+  - Testing
+  - Test Coverage
+summary: |
+  100% test coverage is a contentious metric! In this piece, Olu explores the
+  impact of pursuing it.
+---
+
+Test coverage is a way of measuring how much of the codebase is covered by
+tests. It usually refers to unit tests or integration tests. There are automated
+ways of measuring it, and these usually present your coverage as a percentage.
+100% test coverage is a contentious issue. At OddBird we often aim for this
+level of test coverage, and after working on a few different projects I thought
+it would be good to assess how I find it and the pros and cons of testing in
+this way.
+
+## **The dimensions of testing**
+
+Through working on this blog post I’ve thought a lot about “dimensions” of
+testing: coverage, ‘manualness’, reliability, maintainability and area.
+‘Manualness’ refers to the degree to which you can automate the tests.
+Reliability refers to whether your tests are ‘flaky’, or unstable between runs,
+and can’t be trusted on a given single run. Maintainability is how easy it is to
+keep your tests consistent with a changing codebase during its lifetime. Area
+refers to what part of your codebase is covered by the testing method you are
+using.
+
+These dimensions mean that 100% unit test coverage is rarely actually covering
+every aspect that it could in a codebase. If there was a visual regression in
+one of our codebases with 100% test coverage, for example, our tests wouldn’t
+record this. Visual regression tests don’t have a coverage metric, and can be
+unwieldy when used on large codebases due to the number of screenshots that have
+to be reviewed, but cover a part of the codebase not easily tested by other
+means.
+
+Recently I’ve been working on polyfills for bleeding edge CSS features, and they
```

Comment:
Could we link to one or some of the "polyfills for bleeding edge CSS features" you've been working on?